lib/libefi/rdwr_efi.c

```diff
@@ -1255,19 +1300,61 @@ efi_use_whole_disk(int fd)
 	efi_label->efi_parts[resv_index].p_start += difference;
 	efi_label->efi_last_u_lba = efi_label->efi_last_lba - nblocks;
 
-	rval = efi_write(fd, efi_label);
-	if (rval < 0) {
-		if (efi_debug) {
-			(void) fprintf(stderr,
-			    "efi_use_whole_disk:fail to write label, rval=%d\n",
-			    rval);
+#ifdef BLKPG_RESIZE_PARTITION
```

Comment:
Can you also add a comment for the motivation of doing that here rather than letting `vdev_disk_open()` take care of it?

Also, the caller of `efi_use_whole_disk()` has a comment that would need to be updated:
```
	 * Also, we don't call efi_rescan() - that would just return EBUSY.
	 * The module will do it for us in vdev_disk_open().
	 */
	error = efi_use_whole_disk(fd);
```